### PR TITLE
Added doPrecast to Chestmania; Added two new features

### PR DESCRIPTION
**Pull request by Nishimura-Katsuo**
New Config.AggressiveCloak option for Assassins
New random game name/password option

### CODE REVIEW

File: d2bs/kolbot/libs/config/Assassin.js [hunk 1]
```diff
@@ -482,6 +482,7 @@ function LoadConfig() {
 	Config.UseBoS = false; // Set to true to use Burst of Speed prebuff. TODO: Casting in town + UseFade compatibility
 	Config.UseVenom = false; // Set to true to use Venom prebuff. Set to false if you don't have the skill and have Arachnid Mesh - it will cause connection drop otherwise.
 	Config.UseCloakofShadows = true; // Set to true to use Cloak of Shadows while fighting. Useful for blinding regular monsters/minions.
+	Config.AggressiveCloak = false; // Get into Cloak range on purpose, otherwise only casts if already close enough
```
Comment (noah-) [inline d2bs/kolbot/libs/config/Assassin.js]: change comment to something like 
"Move into Cloak range or cast if already close"

File: d2bs/kolbot/libs/common/Attacks/Assassin.js [hunk 2]
```diff
@@ -30,6 +30,16 @@ var ClassAttack = {
 			untimedSkill = -1;
 
 		index = ((unit.spectype & 0x7) || unit.type === 0) ? 1 : 3;
+
+		// Cloak of Shadows (Aggressive) - can't be cast again until previous one runs out and next to useless if cast in precast sequence (won't blind anyone)
+		if (Config.AggressiveCloak && Config.UseCloakofShadows && me.getSkill(264, 1) && !me.getState(121) && !me.getState(153)) {
+			if(getDistance(me, unit) < 20) {
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Attacks/Assassin.js]: fix format

File: d2bs/kolbot/D2BotLead.dbj [hunk 3]
```diff
@@ -242,13 +242,14 @@ function timer(tick) {
 	return " (" + min + ":" + sec + ")";
 }
 
-function randomString(len) {
+function randomString(len,useNumbers = false) {
```
Comment (noah-) [inline d2bs/kolbot/D2BotLead.dbj]: fix format

File: d2bs/kolbot/D2BotLead.dbj [hunk 4]
```diff
@@ -242,13 +242,14 @@ function timer(tick) {
...
 	var i,
 		rval = "",
-		letters = "abcdefghijklmnopqrstuvwxyz";
-
+		letters =  (useNumbers ? "abcdefghijklmnopqrstuvwxyz0123456789" : "abcdefghijklmnopqrstuvwxyz");
+	if(!len)
```
Comment (noah-) [inline d2bs/kolbot/D2BotLead.dbj]: fix format

File: d2bs/kolbot/D2BotLead.dbj [hunk 5]
```diff
@@ -242,13 +242,14 @@ function timer(tick) {
...
+		len=Math.floor(5+Math.random()*10);
```
Comment (noah-) [inline d2bs/kolbot/D2BotLead.dbj]: fix format

**Comment by Nishimura-Katsuo** [comment]
Submitting new pull request

**Pull request closed without merging**